core/tasks.py

```diff
@@ -64,14 +65,24 @@ def jsonDecodeObjectHook(obj):
 
 
 _gaeApp = os.environ.get("GAE_APPLICATION")
-regionMap = {  # FIXME! Can we even determine the region like this?
-    "h": "europe-west3",
-    "e": "europe-west1"
-}
+
 queueRegion = None
 if _gaeApp:
-    regionPrefix = _gaeApp.split("~")[0]
-    queueRegion = regionMap.get(regionPrefix)
+
+    try:
+        headers = {"Metadata-Flavor": "Google"}
+        r = requests.get("http://metadata.google.internal/computeMetadata/v1/instance/zone", headers=headers)
+        # r.text should be look like this "projects/(project-number)/zones/(zone)-1"
+        # like so "projects/1234567890/zones/europe-west3-1"
+        queueRegion = "-".join(r.text.split("/")[-1].split("-")[:2])
```

Comment:
You can save the join:
```suggestion
        queueRegion = r.text.split("/")[-1].rsplit("-", 1)[0]
```